repo2docker/app.py

```diff
@@ -381,6 +395,21 @@ def initialize(self):
 
         self.run_cmd = args.cmd
 
+        if args.all_ports and not self.run:
+            print('To publish user defined port mappings, the container must also be run')
+            sys.exit(1)
+
+        if args.port and not self.run:
+            print('To publish user defined port mappings, the container must also be run')
+            sys.exit(1)
+
+        if args.port and not self.run_cmd:
+            print('To publish user defined port mapping, user must specify the command to run in the container')
+            sys.exit(1)
+
+        self.port = validate_and_generate_port_mapping(args.port)
```

Comment:
This should be self.ports rather than self.port, no?

Reply:
my bad ... will change